kernel/src/platform/mod.rs

```diff
@@ -100,6 +101,18 @@ pub trait Chip {
     unsafe fn atomic<F, R>(&self, f: F) -> R
     where
         F: FnOnce() -> R;
+
+    /// Print out chip state (system registers) to a supplied
+    /// writer. This does not print out the execution context
+    /// (data registers), as this depends on how they are stored;
+    /// that is implemented by
+    /// `syscall::UserspaceKernelBoundary::print_context`.
+    /// This also does not print out a process memory state,
+    /// that is implemented by `process::Process::print_memory_map`.
+    /// The MPU state is printed by the MPU's implementation of
+    /// the Display trait.
+    /// Used by panic.
+    unsafe fn print_state(&self, writer: &mut dyn Write);
```

Comment:
This doesn't seem like a function that should be marked `unsafe`. It should only read state, and display it. Are there memory safety concerns there?

Reply:
The original one was unsafe so I assumed it was for a good reason and made this unsafe too. It does access system registers, after all. You might not want to leak that state.

Reply:
I'm interested in @alevy's take, but accessing system registers, if we want to protect that, seems like something a capability would restrict. The functions were marked unsafe, but they were set that way long before we had this notion of restricting `unsafe` to just memory and type safety issues. This is the first time we have actually reviewed these functions and this interface.